lib/js/jslib.js

```diff
@@ -1404,9 +1404,66 @@ const LINKS = new (function() {
 const _debug = DEBUG.debug;
 const debug = LINKS.kify(DEBUG.debug);
 
+function _regexCompile(expr){ 
+  switch(expr._label){
+    case "Seq": {
+      const regexes = LINKEDLIST.toArray(expr._value);
+      let acc = "";
+      for(let regex of regexes){
+        acc = acc + _regexCompile(regex);
+      }    
+      return acc;
+    }
+    case "StartAnchor":{
+      return "^";
+    }
+    case "Repeat":{
+      let op = undefined;
+      switch(expr._value[1]._label){
+        case "Plus": op = "+"; break;
+        case "Star": op = "*"; break;
+        case "Question": op = "?"; break;
+        default: throw("Error: Unrecognised Operation");
+      }
+      const regex = _regexCompile(expr._value[2]);
+      return regex + op;
+    }
+    case "EndAnchor":{
+      return "$";
+    }
+    case "Range":{
+      const first = expr._value[1]._c;
+      const second = expr._value[2]._c;
+      return "[" + first + "-" + second + "]";
+    }
+    case "Simply": {
+      return expr._value;
+    }
+    case "Any":{
+      return ".";
+    }
+    case "Alternate":{
+      const first = _regexCompile(expr._value[1]);
+      const second = _regexCompile(expr._value[2]);
+      return first + "|" + second;
+    }
+    case "Group":{
+      return "(" + _regexCompile(expr._value) + ")";
+    }
+    case "Quote":{
+      return "\\" + _regexCompile(expr._value);
+    }
+    case "Replace":{
+      throw("Error: \"Replace\" not implemented");
+    }
+    default: throw("Unrecognised Tag in Regex Expression");
+  }
+  return null;
```

Comment:
Maybe add a comment here `// Can never be reached`. And use `_error` rather than `throw`.

Reply:
Fixed.